packages/suite/src/views/wallet/coinmarket/sell/components/SellForm/Inputs/index.tsx

```diff
@@ -84,14 +70,20 @@ const Inputs = () => {
         setValue(FIAT_INPUT, '', { shouldDirty: true });
         setValue(OUTPUT_AMOUNT, '', { shouldDirty: true });
         clearErrors([FIAT_INPUT, CRYPTO_INPUT]);
-        setActiveInput(CRYPTO_INPUT);
         composeRequest(CRYPTO_INPUT);
-    }, [clearErrors, composeRequest, setActiveInput, setValue]);
+    }, [clearErrors, composeRequest, setValue]);
 
     const isBalanceZero = tokenData
         ? isZero(tokenData.balance || '0')
         : isZero(account.formattedBalance);
 
+    // Trigger validation once amountLimits are loaded after first submit
+    useEffect(() => {
```

Comment:
maybe revert it back – looks more complex to get it right...

Reply:
I believe I got it right with the changes to `useCoinmarketSellForm` in 7ad95de081aabfd0379b7d470a92f82307867f37. There is also a fix in `validation.ts` for [this bug](https://satoshilabs.slack.com/archives/C031LD98L9K/p1688567490380339?thread_ts=1688480384.320529&cid=C031LD98L9K).

f136b9548df266bbd2b2750a3dd0c8441f9e0103 fixes a console warning (h3 being a child of h1) which I came across while fixing the above (not related to react-hook-form).

Reply:
works better now! I have noticed 1 more thing, unlike in the send and exchange forms, an invalid value is persisted in the sell form and not later validated when you come beck. Could you prevent it from bein persisted if it's incorrect like in the other forms since you have touched it extensively already?

Reply:
Looks like this has always been broken. Fixed by checking whether a transaction `isComposing`. 0138d98b6428da21e068af0c1c55b91ca3f6f28f